src/main/java/com/minecolonies/coremod/colony/buildings/BuildingFarmer.java

```diff
@@ -259,17 +292,28 @@ public void onDestroyed()
                 tempField.setTaken(false);
                 tempField.setOwner("");
                 @NotNull final ScarecrowTileEntity scarecrowTileEntity = (ScarecrowTileEntity) getColony().getWorld().getTileEntity(field.getID());
-                getColony().getWorld()
-                  .notifyBlockUpdate(scarecrowTileEntity.getPos(),
-                    getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
-                    getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
-                    BLOCK_UPDATE_FLAG);
-                scarecrowTileEntity.setName(LanguageHandler.format("com.minecolonies.coremod.gui.scarecrow.user",
-                  LanguageHandler.format("com.minecolonies.coremod.gui.scarecrow.user.noone")));
+
+                if (getColony() != null && getColony().getWorld() != null)
+                {
+                    getColony().getWorld()
+                      .notifyBlockUpdate(scarecrowTileEntity.getPos(),
+                        getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
+                        getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
+                        BLOCK_UPDATE_FLAG);
+                    scarecrowTileEntity.setName(LanguageHandler.format("com.minecolonies.coremod.gui.scarecrow.user",
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "com.minecolonies.coremod.gui.scarecrow.user" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1192)